### PR TITLE
fix go.mod to use tagged releases

### PR DESCRIPTION
**Pull request by liamsi**
quick fix for #606

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -255,5 +255,5 @@ replace (
 	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v0.44.1-celestia
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/libp2p/go-libp2p-pubsub v0.5.7-0.20211029175501-5c90105738cf => github.com/celestiaorg/go-libp2p-pubsub v0.5.7-0.20220325112944-d33a3e5e13d5
```
Comment (liamsi) [inline go.mod:257]: @Wondertan should I also update this dependency?

File: go.mod [hunk 2]
```diff
@@ -255,5 +255,5 @@ replace (
...
-	github.com/tendermint/tendermint v0.34.14 => github.com/celestiaorg/celestia-core v0.34.14-celestia.0.20220228062426-721562a3544d
+	github.com/tendermint/tendermint v0.34.14 => github.com/celestiaorg/celestia-core v0.34.15-celestia
```
Comment (liamsi) [inline go.mod:258]: This should also make the code duplication in #529 unnecessary.

File: go.mod (hunk 1 above)
Comment (Wondertan) [reply go.mod:257]: We can tag things on our pubsub fork just to be consistent with tagging. Could you do it?

Comment (liamsi) [reply go.mod:257]: Yes

Comment (Wondertan) [reply go.mod:257]: The branch: https://github.com/celestiaorg/go-libp2p-pubsub/tree/old-datastore

Comment (liamsi) [reply go.mod:257]: https://github.com/celestiaorg/go-libp2p-pubsub/releases/tag/v0.5.6-celestia

**Review by Wondertan — APPROVED** [review]
Thanks!

**Comment by liamsi** [comment]
unfortunately this is blocked on `TestSyncLightWithBridge`, `TestSyncFullWithBridge`, `TestSyncLightWithFull`, `TestSyncLightWithTrustedPeers` consistently failing.

**Comment by liamsi** [comment]
OK, actually. Same tests above also fail here: https://github.com/celestiaorg/celestia-node/pull/612
So these failures are orthogonal to this PR. Though I think we must fix these tests asap instead of ignoring them.

**Pull request merged**